### PR TITLE
[Transaction Controller Migration] Adopt rename transactionHash property to hash

### PR DESCRIPTION
**Pull request by vinistevam**
## **Description**

Update the `@metamask/transaction-controller` patch to:

- Rename `transactionHash` to `hash` on transaction metadata.

## **Manual Testing Steps**

Full transaction regression.

## **Pre-merge Author Checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained:
  - [x] What problem this PR is solving.
  - [x] How this problem was solved.
  - [x] How reviewers can test my changes.
- [x] I’ve indicated what issue this PR is linked to: Fixes #???
- [x] I’ve included tests if applicable.
- [x] I’ve documented any added code.
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: patches/@metamask+transaction-controller+4.0.1.patch [hunk 1]
```diff
@@ -2013,6 +2029,170 @@ index 0000000..79f6512
...
++++ b/node_modules/@metamask/transaction-controller/dist/external-transactions.js
+@@ -0,0 +1,36 @@
++"use strict";
++Object.defineProperty(exports, "__esModule", { value: true });
++exports.validateConfirmedExternalTransaction = void 0;
++// These utility functions are exclusively used by `confirmExternalTransaction` method in controller
++const rpc_errors_1 = require("@metamask/rpc-errors");
++const types_1 = require("./types");
++/**
++ * Validates the external provided transaction meta.
```
Comment (matthewwalsh0) [inline patches/@metamask+transaction-controller+4.0.1.patch:2062]: Just to clarify for audit sake, but is this patching not just the hash changes specifically but all non-breaking changes up to that point also?

File: patches/@metamask+transaction-controller+4.0.1.patch [hunk 2]
```diff
@@ -2013,6 +2029,170 @@ index 0000000..79f6512
...
++    transactionMeta.history = [snapshot];
++}
++exports.addInitialHistorySnapshot = addInitialHistorySnapshot;
++/**
++ * Compares and adds history entry to the provided transactionMeta history.
++ *
++ * @param transactionMeta - TransactionMeta to add history entry to.
++ * @param note - Note to add to history entry.
++ */
++function updateTransactionHistory(transactionMeta, note) {
```
Comment (matthewwalsh0) [inline patches/@metamask+transaction-controller+4.0.1.patch:2144]: As this appears to be the history support, do we also need to set `disableHistory` to true in the engine?

File: app/store/migrations.test.js [hunk 3]
```diff
@@ -983,4 +983,88 @@ describe('Redux Persist Migrations', () => {
...
+                  transactionHash: '0x456',
+                  otherProperty: 'otherValue',
+                },
+              ],
+            },
+          },
+        },
+      };
+
+      const migration = migrations[23];
```
Comment (matthewwalsh0) [inline app/store/migrations.test.js:1008]: Note to update the name of the `describe` block plus these indexes in each test.

Comment (vinistevam) [reply app/store/migrations.test.js]: Same for the whole test https://github.com/MetaMask/metamask-mobile/pull/7344/commits/36ef1f52bb9b89850ed421f449f2ddb19beda69c

File: app/store/migrations.js [hunk 4]
```diff
@@ -760,6 +760,23 @@ export const migrations = {
 
     return state;
   },
+  23: (state) => {
```
[the remark this answers is not in the document]
Comment (vinistevam) [reply app/store/migrations.js]: Bumped to 26 https://github.com/MetaMask/metamask-mobile/pull/7344/commits/36ef1f52bb9b89850ed421f449f2ddb19beda69c

File: patches/@metamask+transaction-controller+5.0.0.patch [hunk 5]
```diff
@@ -11,6 +11,400 @@ index 0000000..550de56
 +* Run "yarn build" in the core monorepo
 +* Run "yarn patch:tx <core-directory>" in the mobile repo
 +* Once the new patch is merged, add your changes to: refactor/transaction-controller-patch-mobile-5
+diff --git a/node_modules/@metamask/transaction-controller/dist/EtherscanRemoteTransactionSource.d.ts b/node_modules/@metamask/transaction-controller/dist/EtherscanRemoteTransactionSource.d.ts
+new file mode 100644
+index 0000000..d7d701f
+--- /dev/null
++++ b/node_modules/@metamask/transaction-controller/dist/EtherscanRemoteTransactionSource.d.ts
```
Comment (matthewwalsh0) [inline patches/@metamask+transaction-controller+5.0.0.patch:18]: I'm concerned by these paths as they're not using the new `helpers` and `utils` directories but those paths are still in the patch.

I've seen similar issues when re-building core without first deleting the `dist` folder as it ends up with the old and new files in.

Comment (vinistevam) [reply patches/@metamask+transaction-controller+5.0.0.patch:18]: Let me try again, but first delete the `dist`  first just to make sure.

Comment (vinistevam) [reply patches/@metamask+transaction-controller+5.0.0.patch]: Done https://github.com/MetaMask/metamask-mobile/pull/7344/commits/be1cb5ed055a40308abe602f52f0258542af8d53

**Review by matthewwalsh0 — APPROVED** [review]
